src/Middleware/RequestDecompression/src/DeflateDecompressionProvider.cs

```diff
@@ -13,6 +14,221 @@ internal sealed class DeflateDecompressionProvider : IDecompressionProvider
     /// <inheritdoc />
     public Stream GetDecompressionStream(Stream stream)
     {
-        return new DeflateStream(stream, CompressionMode.Decompress, leaveOpen: true);
+        return new ZLibOrDeflateStream(stream);
```

Comment:
is it OK to read the byte here, so you can return DeflateStream or ZLibStream directly and eliminate ZLibOrDeflateStream? 

it seems to me it might be, since after GetDecompressionStream(..) the stream is immediately passed to be used, without any intervening waits on other things. but I don't know this code ofc.
https://github.com/danmoseley/aspnetcore/blob/658c7e81ece5b61d772abff859d211102b0d759e/src/Middleware/RequestDecompression/src/RequestDecompressionMiddleware.cs#L68

Reply:
GetDecompressionProvider_SupportedContentEncoding_ReturnsProvider eg would need to be tweaked to ensure the HttpRequest had a tiny body.